crypto/fips_callback_test.cc

```diff
@@ -106,15 +108,55 @@ TEST(FIPSCallback, DRBGRuntime) {
   uint8_t buf[10];
   if (broken_runtime_test != nullptr && strcmp(broken_runtime_test, "CRNG" ) == 0) {
     ASSERT_FALSE(RAND_bytes(buf, sizeof(buf)));
-    // TODO: make AWS_LC_FIPS_error update a new global state so FIPS_mode returns 0
-    ASSERT_EQ(1, FIPS_mode());
     ASSERT_EQ(1, failure_count);
   } else {
     // BORINGSSL_FIPS_BREAK_TEST has not been set and everything should work
     ASSERT_TRUE(RAND_bytes(buf, sizeof(buf)));
-    ASSERT_EQ(1, FIPS_mode());
     ASSERT_EQ(0, failure_count);
   }
+  ASSERT_EQ(1, FIPS_mode());
```

Comment:
Duplicate test to L107. I think keeping the earlier FIPS assertion test makes more sense though.

Reply:
I wanted this here to verify the behavior: that `FIPS_mode` doesn't change.

Reply:
Should we also include an early FIPS assertion test for `TEST(FIPSCallback, PowerOnTests)` to verify this behavior for those tests as well?